src/window.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class WindowService {
+
+  get nativeWindow(): any {
+    return _window();
+  }
+}
+
+function _window(): any {
+  // return the global native browser window object
+  return typeof window !== 'undefined' ? window : undefined;
+}
```

Comment:
@tjoskar The point of `WindowService` is to provide it in `ngModule`, and then inject it in the directive constructor so it can be recognized as a service. what you are doing here is getting window  object by calling a static function in a static class which is the same as just using `window` directly